### PR TITLE
Sort clients and servers on admin dashboard

### LINKED ISSUE #915: admin dashboard: sort servers and clients alphabetically
I think this is a relatively small change, but it would be great to sort the routers and clients that show up on the admin dashboard alphabetically, so that they load in a consistent order each time. Right now I think it's just hash ordering. Here's an example:

screen shot 2017-01-10 at 2 19 18 pm

### PR DESCRIPTION
**Pull request by klingerf**
Sorts clients and servers on the admin dashboard alphabetically by their label. The values are pulled from a map, and without sorting they show up in arbitrary order. After the page loads, as new clients are added they are also inserted alphabetically into the page, and the coloring of the existing clients is redrawn accordingly.

As part of this branch, I'm also sorting routers in the order that they're defined in the config file, and I'm removing the old metrics handler and all of its assets, since that page is no longer accessible in the admin dashboard.

Fixes #915.

### CODE REVIEW

**Comment by rmars** [comment]
Looks awesome, thanks for doing this.

I think we'll also need to add something in `router_clients.js` - when new clients show up they'll just be appended to the dom... then the clients won't appear in sorted order.  

So we'd need to clear and reappend all the `client-container`s. This could be expensive if there's a lot of clearing and reappending going on so maybe that's not worth it, since you could always refresh to get a sorted page. (Though maybe we don't want people to have to do this).


example:
add a bunch of clients
screen shot 2017-01-11 at 5 13 45 pm

need to refresh to resort:
screen shot 2017-01-11 at 5 14 39 pm

**Comment by rmars** [comment]
I think we'll also need to change `assignColorsToClients` in `router-clients.js`(my fault for making it order-dependent) - the reordering is causing color mapping issues, were more than 1 client is assigned the same color, even though we have more colors available:

screen shot 2017-01-11 at 5 21 04 pm

**Comment by adleong** [comment]
@rmars is is possible to insert the new client into the correct position in the dom instead of clearing and re-appending?

**Comment by rmars** [comment]
yeah! good point! we can use $.insertBefore and the like to just do inserts

**Comment by klingerf** [comment]
@rmars Thanks for the suggestion! I added client sort-on-update in 7504903c. I also removed a few things that appeared to be unused, but please have a look to make sure I didn't break anything.

**Comment by adleong** [comment]
Does this mean that when new clients are added, existing clients might change color?

**Comment by klingerf** [comment]
@adleong That's correct.

File: admin/src/main/resources/io/buoyant/admin/js/src/logger-overrides.js [hunk 1]
```diff
@@ -12,6 +12,7 @@ $(function() {
   '        <div id="navbar" class="collapse navbar-collapse">' +
   '          <ul class="nav navbar-nav">' +
   '            <li><a href="../delegator">dtab</a></li>' +
+  '            <li><a href="../requests">requests</a></li>' +
```
Comment (adleong) [inline admin/src/main/resources/io/buoyant/admin/js/src/logger-overrides.js]: ahh, good catch.  thanks for fixing!

File: admin/src/main/resources/io/buoyant/admin/js/src/router_clients.js [hunk 2]
```diff
@@ -13,12 +13,13 @@ define([
...
 
     function assignColorsToClients(colors, clients) {
-      var colorIdx = 0;
+      var clientMapping = {};
 
-      return _.reduce(clients, function(clientMapping, client) {
-        clientMapping[client.label] = colors[colorIdx++ % colors.length];
-        return clientMapping;
-      }, {});
+      _(clients).sortBy('label').forEach(function(client, idx) {
```
Comment (siggy) [inline admin/src/main/resources/io/buoyant/admin/js/src/router_clients.js]: there's a `return _.sortBy(clients, 'label');` on line 204 of `routers.js`. is this sorting twice?

File: admin/src/main/resources/io/buoyant/admin/js/src/router_clients.js [hunk 3]
```diff
@@ -70,15 +79,15 @@ define([
           routerClient.updateColors(clientToColor);
         });
 
-        var expandClients = shouldExpandClients(addedClients.length + routerClients.length);
+        expandClients = shouldExpandClients(routers.clients(routerName).length);
```
Comment (siggy) [inline admin/src/main/resources/io/buoyant/admin/js/src/router_clients.js]: nit: i see `routers.clients(routerName)` repeated (and also assignned to `clients`), consider DRY up.

File: admin/src/main/resources/io/buoyant/admin/js/src/logger-overrides.js (hunk 1 above)
Comment (siggy) [inline admin/src/main/resources/io/buoyant/admin/js/src/logger-overrides.js]: consider a comment here and/or in `AdminHandler.scala` that the nav's should stay in sync.

Comment (rmars) [inline admin/src/main/resources/io/buoyant/admin/js/src/logger-overrides.js]: (I'm working on that right now as part of #505!)

**Comment by klingerf** [comment]
@adleong Ok, per discussion, have updated the behavior to append new clients and preserve the color of existing clients.

**Comment by adleong** [comment]
👏 👏 👏  thank you, @klingerf!  Sorry for the runaround

**Pull request merged**